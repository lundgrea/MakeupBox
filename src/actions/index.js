export const handleErrors = error => ({
  type: "HANDLE_ERROR",
  error
})

export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
})

export const getResults = (results) => ({
  type: 'GET_RESULTS',
  results
})

export const setResponses = response => ({
  type: 'SET_RESPONSES',
  response
})