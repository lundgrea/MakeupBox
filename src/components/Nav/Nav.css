nav {
  max-height: 75vh;
  width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0px 0px 0px 0px;
    padding: 10px 10px 10px 10px;
    background-color: black;
    color: white;
}

.logo{
  color: white;
}
.nav-menu {
  margin: 10px 50px 10px 10px;
  font-size: 22px;
  color: white;
}

h1 {
  font-size: 60px;
  margin: 10px 10px 10px 20px;
}

.tagline {
  font-size: 16px;
  margin: 0px 0px 0px 20px;
}