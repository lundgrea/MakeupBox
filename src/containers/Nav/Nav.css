nav {
  max-height: 75vh;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0px 0px 0px 0px;
  padding: 10px 10px 10px 10px;
  background-color: #242B28;
  color: white;
}

header {
  display: flex;
}

.logo{
  color: white;
}

.nav-menu {
  margin: 10px 50px 10px 10px;
  font-size: 22px;
  color: white;
}

.nav-menu:hover {
color: #8EA99E
}

h1 {
  font-size: 60px;
  margin: 10px 10px 10px 10px;
}

.tagline {
  font-size: 16px;
  margin: 0px 0px 0px 30px;
}

.logo-image {
  height: 50px;
  margin: 20px 0px 0px 20px;
  filter:invert(100%) sepia(1%) saturate(50%) hue-rotate(1deg);
}