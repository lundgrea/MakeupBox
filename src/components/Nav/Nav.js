import React from 'react';
import './Nav.css'
import { NavLink } from 'react-router-dom';

export const Nav = () => {

const emptyFields = () => {
  console.log('in the empty')
}

return (
  <nav>
    <h1>makeupBOX</h1>
    <NavLink to='/' onClick={emptyFields} className='nav'> Start Over </NavLink>
   </nav>
   )
}

export default Nav

