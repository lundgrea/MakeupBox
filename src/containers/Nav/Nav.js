import React from 'react';
import './Nav.css'
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { clearResults, clearResponses } from '../../actions';
import PropTypes from 'prop-types';


export const Nav = (props) => {

const emptyFields = () => {
  props.clearResults();
  props.clearResponses();
};

return (
  <nav>
    <div className='logo-lockup'>
      <header>
        <img className='logo-image' alt='logo' src='https://image.flaticon.com/icons/png/512/2081/2081669.png'/>
        <Link to='/'><h1 className='logo'>TAKEUP</h1></Link>
      </header>
      <h2 className='tagline'>a makeup exploration tool for men</h2>
    </div>
    <NavLink to='/' onClick={emptyFields} className='nav-menu'> start over </NavLink>
   </nav>
   );
};

export const mapDispatchToProps = dispatch => ({
  clearResults: () => dispatch(clearResults()),
  clearResponses: () => dispatch(clearResponses())
});


export default connect(null, mapDispatchToProps)(Nav);


Nav.propTypes = {
  clearResponses:PropTypes.func.isRequired,
  clearResults:PropTypes.func.isRequired,
};
