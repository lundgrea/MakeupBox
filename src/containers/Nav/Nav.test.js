import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapDispatchToProps } from './Nav';
import { clearResults, clearResponses } from '../../actions'

let wrapper;

beforeEach(()=> {
  wrapper = shallow(<Nav />);
})

describe('Nav', () => {
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with a clearResults action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = clearResults();
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.clearResults()
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

  it('should call dispatch with a clearResponses action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = clearResponses();
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.clearResponses()
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)

  });

});
