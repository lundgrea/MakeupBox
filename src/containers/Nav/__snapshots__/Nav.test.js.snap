// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav>
  <div
    className="logo-lockup"
  >
    <header>
      <img
        alt="logo"
        className="logo-image"
        src="https://image.flaticon.com/icons/png/512/2081/2081669.png"
      />
      <Link
        to="/"
      >
        <h1
          className="logo"
        >
          TAKEUP
        </h1>
      </Link>
    </header>
    <h2
      className="tagline"
    >
      a makeup exploration tool for men
    </h2>
  </div>
  <NavLink
    className="nav-menu"
    onClick={[Function]}
    to="/"
  >
     start over 
  </NavLink>
</nav>
`;
